### PR TITLE
Replace Nix grammar

### PR DESCRIPTION
**Pull request by sambacha**
Change the submodule that linguist uses for Nix highlighting to an updated grammar syntax. Currently the one in use has issues and is not maintained.

## Description

Update to latest Nix grammar
Change submodule path, should be a drop in (hopefully).

## Checklist:

- [x] **I am changing the source of a syntax highlighting grammar**
  - Old: https://github.com/wmertens/sublime-nix
  - New: https://github.com/sambacha/nix-linguist

replaces pr: https://github.com/github/linguist/pull/5950

### CODE REVIEW

File: vendor/licenses/git_submodule/nix-linguist.dep.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+---
+name: nix-linguist
+version: 83d5d25e1ad087ea8e899194357fc8bdd8da3ee2
+type: git_submodule
+homepage: https://github.com/sambacha/nix-linguist.git
+license: other
```
Comment (lildude) [inline vendor/licenses/git_submodule/nix-linguist.dep.yml:6]: Reviewed the license and can confirm this is MIT. Feel free to change this:

```suggestion
license: mit
```

**Comment by sambacha** [comment]
thanks @lildude , wonder why it reported as `other` ? Should there be some SPDX identifier? 

Much appreciated + many thanks!

File: vendor/licenses/git_submodule/nix-linguist.dep.yml (hunk 1 above)
Comment (sambacha) [reply vendor/licenses/git_submodule/nix-linguist.dep.yml]: 💯

**Comment by lildude** [comment]
> thanks @lildude , wonder why it reported as other ? Should there be some SPDX identifier?

I suspect it's because you've modified the license file so now it's not quite what https://github.com/github/licensed expects.

**Comment by sambacha** [comment]
> > thanks @lildude , wonder why it reported as other ? Should there be some SPDX identifier?
> 
> I suspect it's because you've modified the license file so now it's not quite what https://github.com/github/licensed expects.

Thanks. Is there any outstanding blockers to this? Cheers

**Comment by sambacha** [comment]
> Oh wait. You need to merge master into your fork as you've not given maintainers permissions to do it.

Oops, thanks. Just did that.

**Comment by sambacha** [comment]
Thanks for your help! Much appreciated!

**Comment by SuperSandro2000** [comment]
I think this change rolled out to github.com today and with it attr keys are now colored blue, too like most other things. You can see that in https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/misc/hello/default.nix

For me personally this made the syntax highlighting a lot worse and less helpful and by that creating an accessibility issue.

**Comment by lildude** [comment]
Unfortunately not @SuperSandro2000. This change rolled out a long time ago. What rolled out yesterday was switching to using the upstream [tree-sitter grammar](https://github.com/nix-community/tree-sitter-nix.git) by the team that manages the highlighting engine (completely independent of Linguist) as this grammar was causing stack exhaustion issues.

I'll pass this comment on to them, but ultimately this would need to be raised with the upstream grammar

**Comment by SuperSandro2000** [comment]
Thanks for the fast answer and the helpful link. I raised an issue in the repo https://github.com/nix-community/tree-sitter-nix/issues/40

**Pull request merged**